test/e2e/apps/disruption.go

```diff
@@ -417,6 +419,78 @@ var _ = SIGDescribe("DisruptionController", func() {
 		framework.ExpectNoError(err) // the eviction is now allowed
 	})
 
+	ginkgo.Context("Reacting to DaemonSet pods", func() {
+		ginkgo.BeforeEach(func() {
+			ginkgo.By(fmt.Sprintf("Create a DaemonSet %q.", defaultName))
+			ds, err := cs.AppsV1().DaemonSets(ns).Create(context.TODO(), newDaemonSet(defaultName, WebserverImage, defaultLabels), metav1.CreateOptions{})
+			framework.ExpectNoError(err)
+
+			ginkgo.By("Check that daemon pods launch on every node of the cluster.")
+			err = e2edaemonset.CheckDaemonStatus(context.TODO(), f, defaultName)
+			framework.ExpectNoError(err)
+			err = wait.PollImmediateWithContext(context.TODO(), dsRetryPeriod, dsRetryTimeout, checkRunningOnAllNodes(f, ds))
+			framework.ExpectNoError(err)
+		})
+
+		ginkgo.AfterEach(func() {
+			ginkgo.By("Clean up.")
+			err := cs.AppsV1().DaemonSets(ns).Delete(context.TODO(), defaultName, metav1.DeleteOptions{})
+			framework.ExpectNoError(err)
+		})
+
+		// Test PDB reacting to daemonset pod conditions
+		pdbCases := []struct {
+			createPDB func(int)
+			name      string
+		}{
+			{
+				name: "minAvailable",
+				createPDB: func(Npods int) {
+					createPDBMinAvailableOrDie(context.TODO(), cs, ns, defaultName, intstr.FromInt(Npods-1), defaultLabels)
+				},
+			},
+
+			{
+				name: "maxUnavailable",
+				createPDB: func(_ int) {
+					createPDBMaxUnavailableOrDie(context.TODO(), cs, ns, defaultName, intstr.FromString("10%"))
+				},
+			},
+		}
+
+		for _, pdbCase := range pdbCases {
+			ginkgo.It("should account pods of a DaemonSet in case of "+pdbCase.name,
+				func() {
+					podList := listDaemonPods(context.TODO(), cs, ns, defaultLabels)
+					Npods := len(podList.Items)
+
+					ginkgo.By("Creating PDB for " + strconv.Itoa(Npods) + " pods.")
+					pdbCase.createPDB(Npods)
+
+					ginkgo.By("Set a daemon pod's phase to 'Failed'.")
```

Comment:
if you run this on a cluster with more than 10 nodes while using maxUnavailable 10%, downing one pod will not be enough for the assertions to succeed.